lib/iris/cube.py

```diff
@@ -307,10 +341,6 @@ def _extract_and_merge(cubes, constraints, strict, merge_unique=False):
                 if sub_cube is not None:
                     cube_list.append(sub_cube)
 
-        if merge_unique is not None:
```

Comment:
I'm a little nervous that there may have been a future plan for this, but I suppose un-used code is always superfluous at the end of the day?

Reply:
Yes, this code has become redundant, i.e. never called, at some point.
I suspect, from the name, that `_extract_and_merge` used to be called by CubeList.merge also, but no longer is ?
I'm just happy to see the scope of this reduced, as it is all a bit confusing !!

Probably this whole implementation can be further simplified, but I just didn't want to overhaul the whole thing here+now as it would further complicate this PR.

Reply:
> _extract_and_merge used to be called by CubeList.merge also,

After some digging I find it goes back to just before Iris v1.0, when we replaced 'load' and 'load_strict' functions with load / load_cubes / load_raw.
https://github.com/SciTools/iris/pull/78
Prior to that, the load mechanism was exposing 'unique' and 'strict' options, and it all came through `CubeList._extract_and_merge` with the differing options.
I don't think the "merge_unique" option here has been used since then, so it is seriously ancient !